components/remote_settings/src/client.rs

```diff
@@ -1529,11 +1588,111 @@ mod test_new_client {
             }
         });
         let storage = Storage::new(":memory:".into()).expect("Error creating storage");
-        let rs_client =
-            RemoteSettingsClient::new_from_parts("test-collection".into(), storage, api_client);
+        let rs_client = RemoteSettingsClient::new_from_parts(
+            "test-collection".into(),
+            storage,
+            JexlFilter::new(None),
+            api_client,
+        );
         assert_eq!(
             rs_client.get_records(true).expect("Error getting records"),
             Some(records)
         );
     }
 }
+
+#[cfg(test)]
+mod test_filtering_records {
+    use super::*;
+    use serde_json::json;
+
+    #[test]
+    fn test_get_records_filtered_app_version_pass() {
+        let mut api_client = MockApiClient::new();
+        let records = vec![RemoteSettingsRecord {
+            id: "record-0001".into(),
+            last_modified: 100,
+            deleted: false,
+            attachment: None,
+            fields: json!({"foo": "bar"}).as_object().unwrap().clone(),
```

Comment:
Driveby question: in your record representation, do you explicitly exclude `id`, `last_modified`, `deleted`, `attachment` from `fields`?

Reply:
Yeah, because we just test for `filter_expression` here. But maybe we should always have a "real" record representation so it's clear what format the `record` has.